app/javascript/components/bathroom.jsx

```diff
@@ -42,14 +38,14 @@ Bathroom.propTypes = {
   location: PropTypes.shape({
     bathroomCode: PropTypes.string,
   }),
-  loading: PropTypes.bool,
-  error: PropTypes.bool,
 };
 
 Bathroom.defaultProps = {
   location: {},
-  loading: true,
-  error: false,
 };
 
-export default withFragment(Bathroom, { location: getBathroomCode });
+export default compose(
+  renderWhileError(ErrorMessage),
```

Comment:
I don't think it makes sense for the leaf components to be responsible for rendering errors. If there's a GQL error then we want to show that once - presumably where the query is performed - instead of simultaneously in half a dozen places throughout the page.

Reply:
I agree with this sentiment but the goal here is a first pass refactor, not a change of existing functionality.